README.md

```diff
@@ -242,6 +242,214 @@ remote_state = {
+a parent file using the `parent` block:
+
+```hcl
+parent = {
+  path = "../../.terragrunt"
+}
+```
+
+There are a few problems with the simple approach above, so read on before using it!
+
+1. Having to manually manage the file paths to the parent folder is tedious. To solve this problem, you can use the 
+   `find_in_parent_folders` helper. 
+1. If the parent `.terragrunt` file hard-codes the `state_file_id` and `key` settings, then every child would rely on
+   the same lock and write state to the same location. To avoid this problem, you can use the `path_relative_to_parent`
+   helper.
+1. Some of the child `.terragrunt` files may want to override settings from the parent. To do this, see the section
+   on overriding settings from the parent.
+
+Each of these items is discussed next.
+
+### find_in_parent_folders helper
+
+Terragrunt supports the use of a few helper functions using the same syntax as Terraform: `${some_function()}`. One of
+the supported helper functions is `find_in_parent_folders`, which returns the path to the first `.terragrunt` file it
+finds in the parent folders above the current `.terragrunt` file. 
+
+Example:
+
+```hcl
+parent = {
+  path = "${find_in_parent_folders()}"
+}
+```
+
+If you ran this in `qa/my-app/.terragrunt`, this would automatically set `path` to `../../.terragrunt`. You will almost
+always want to use this function, as it allows you to copy/paste the same `.terragrunt` file to all child folders with
+no changes.
+
+### path_relative_to_parent helper
+
+Another helper function supported by Terragrunt is `path_relative_to_parent`, which returns the relative path between 
+the current `.terragrunt` file and the path specified in its `parent` block. For example, in the root `.terragrunt` 
+file, you could do the following:
+ 
+```hcl
+# Configure Terragrunt to use DynamoDB for locking
+lock = {
+  backend = "dynamodb"
+  config {
+    state_file_id = "${path_relative_to_parent()}"
+  }
+}
+
+# Configure Terragrunt to automatically store tfstate files in an S3 bucket
+remote_state = {
+  backend = "s3"
+  config {
+    encrypt = "true"
+    bucket = "my-bucket"
+    key = "${path_relative_to_parent()}/terraform.tfstate"
+    region = "us-east-1"
+  }
+}
+``` 
+
+Using the configuration above, each child `.terragrunt` file will get a unique path for its `state_file_id` and `key` 
+settings. For example, in `qa/my-app/.terragrunt`, the `state_file_id` will resolve to `qa/my-app` and the `key` will
+resolve to `qa/my-app/terraform.tfstate`.  
+
+You will almost always want to use this helper too. The only time you may want to specify the `state_file_id` or `key` 
+manually is if you moved a child folder. In that case, to ensure it can reuse its old state and lock, you may want to 
+hard-code the `state_file_id` and `key` to the old file path. However, a safer approach would be to move the state 
+files themselves to match the new location of the child folder, as that makes things more consistent!
```

Comment:
Seems like a decent tool. File a bug so we can keep track of this and do it in a follow-up PR.